### PR TITLE
refactor!: orb tools 12 migration

### PR DESCRIPTION
**Pull request by brivu**

### CODE REVIEW

File: .circleci/test-deploy.yml [hunk 1]
```diff
@@ -1,32 +1,29 @@
 version: 2.1
 orbs:
-  aws-elastic-beanstalk: circleci/aws-elastic-beanstalk@dev:<<pipeline.git.revision>>
   aws-cli: circleci/aws-cli@3.1
```
Comment (EricRibeiro) [inline .circleci/test-deploy.yml:3]: We still leave the orb defined here to illustrate its injection. It is not required but a good to have.

```suggestion
  aws-elastic-beanstalk: {}
  aws-cli: circleci/aws-cli@3.1
```

File: .circleci/test-deploy.yml [hunk 2]
```diff
@@ -82,59 +79,57 @@ workflows:
...
+          application_name: sample_app
           label: test-label
-          app-dir: ./sample_app
+          app_dir: ./sample_app
           requires:
-                - integration-test-deploy-setup
+            - integration-test-deploy-setup
           filters: *filters
       - orb-tools/pack:
           filters: *filters
```
Comment (EricRibeiro) [inline .circleci/test-deploy.yml]: You no longer need `orb-tools/pack` running on every build. It should run only upon release; therefore, it must have the same filters as `orb-tools/publish`. See [here](https://github.com/CircleCI-Public/Orb-Template/blob/main/.circleci/test-deploy.yml#L16) and [here](https://github.com/CircleCI-Public/Orb-Template/blob/main/.circleci/test-deploy.yml#L41) for the best approach.

File: .circleci/test-deploy.yml [hunk 3]
```diff
@@ -82,59 +79,57 @@ workflows:
...
           context: orb-publisher
-          enable-pr-comment: true
-          github-token: GHI_TOKEN
+          enable_pr_comment: true
+          github_token: GHI_TOKEN
           filters:
             branches:
               ignore: /.*/
             tags:
               only: /^v[0-9]+\.[0-9]+\.[0-9]+$/
```
Comment (EricRibeiro) [inline .circleci/test-deploy.yml]: Change this to match the Orb-Template: https://github.com/CircleCI-Public/Orb-Template/blob/main/.circleci/test-deploy.yml#L51

**Review by EricRibeiro — CHANGES_REQUESTED** [review]
Great work 🔥

**Review by EricRibeiro — CHANGES_REQUESTED** [review]
Great work 🔥

**Review by EricRibeiro — CHANGES_REQUESTED** [review]
Great work 🔥

File: .circleci/test-deploy.yml (hunk 2 above)
Comment (brivu) [reply .circleci/test-deploy.yml]: Done, thank you!

**Review by EricRibeiro — APPROVED** [review]

**Pull request merged**